{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd \n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "import sys\n",
    "import math"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "def progress(count, total, status=''):\n",
    "    bar_len = 60\n",
    "    filled_len = int(round(bar_len * count / float(total)))\n",
    "\n",
    "    percents = round(100.1 * count / float(total), 1)\n",
    "    bar = '=' * filled_len + '-' * (bar_len - filled_len)\n",
    "\n",
    "    sys.stdout.write('[%s] %s%s %s\\r' % (bar, percents, '%', status))\n",
    "    sys.stdout.flush()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "meta = pd.read_csv('../data/combined.tsv', sep='\\t', header=0, index_col=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Virus name</th>\n",
       "      <th>Accession ID</th>\n",
       "      <th>Collection date</th>\n",
       "      <th>Location</th>\n",
       "      <th>Host</th>\n",
       "      <th>Additional location information</th>\n",
       "      <th>Gender</th>\n",
       "      <th>Patient age</th>\n",
       "      <th>Patient status</th>\n",
       "      <th>Passage</th>\n",
       "      <th>Specimen</th>\n",
       "      <th>Additional host information</th>\n",
       "      <th>Lineage</th>\n",
       "      <th>Clade</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>hCoV-19/Mayotte/IPP00846/2021</td>\n",
       "      <td>EPI_ISL_1013099</td>\n",
       "      <td>2021-01-11</td>\n",
       "      <td>Africa / Mayotte / Mamoudzou</td>\n",
       "      <td>Human</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Female</td>\n",
       "      <td>61</td>\n",
       "      <td>Inpatient</td>\n",
       "      <td>Original</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>B.1.160</td>\n",
       "      <td>GH</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>hCoV-19/Mayotte/IPP03105/2021</td>\n",
       "      <td>EPI_ISL_1013300</td>\n",
       "      <td>2021-01-28</td>\n",
       "      <td>Africa / Mayotte / Mamoudzou</td>\n",
       "      <td>Human</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Female</td>\n",
       "      <td>8</td>\n",
       "      <td>Inpatient</td>\n",
       "      <td>Original</td>\n",
       "      <td>Nasopharyngeal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>B.1.351</td>\n",
       "      <td>GH</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>hCoV-19/Mayotte/IPP03108/2021</td>\n",
       "      <td>EPI_ISL_1013301</td>\n",
       "      <td>2021-01-28</td>\n",
       "      <td>Africa / Mayotte / Mamoudzou</td>\n",
       "      <td>Human</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Female</td>\n",
       "      <td>49</td>\n",
       "      <td>Inpatient</td>\n",
       "      <td>Original</td>\n",
       "      <td>Nasopharyngeal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>B.1.351</td>\n",
       "      <td>GH</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>hCoV-19/Mayotte/IPP03109/2021</td>\n",
       "      <td>EPI_ISL_1013302</td>\n",
       "      <td>2021-01-28</td>\n",
       "      <td>Africa / Mayotte / Mamoudzou</td>\n",
       "      <td>Human</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Male</td>\n",
       "      <td>78</td>\n",
       "      <td>Inpatient</td>\n",
       "      <td>Original</td>\n",
       "      <td>Nasopharyngeal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>B.1.351</td>\n",
       "      <td>GH</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>hCoV-19/Mayotte/IPP03110/2021</td>\n",
       "      <td>EPI_ISL_1013303</td>\n",
       "      <td>2021-01-28</td>\n",
       "      <td>Africa / Mayotte / Mamoudzou</td>\n",
       "      <td>Human</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Male</td>\n",
       "      <td>34</td>\n",
       "      <td>Inpatient</td>\n",
       "      <td>Original</td>\n",
       "      <td>Nasopharyngeal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>B.1.351</td>\n",
       "      <td>GH</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                      Virus name     Accession ID Collection date  \\\n",
       "0  hCoV-19/Mayotte/IPP00846/2021  EPI_ISL_1013099      2021-01-11   \n",
       "1  hCoV-19/Mayotte/IPP03105/2021  EPI_ISL_1013300      2021-01-28   \n",
       "2  hCoV-19/Mayotte/IPP03108/2021  EPI_ISL_1013301      2021-01-28   \n",
       "3  hCoV-19/Mayotte/IPP03109/2021  EPI_ISL_1013302      2021-01-28   \n",
       "4  hCoV-19/Mayotte/IPP03110/2021  EPI_ISL_1013303      2021-01-28   \n",
       "\n",
       "                       Location   Host Additional location information  \\\n",
       "0  Africa / Mayotte / Mamoudzou  Human                             NaN   \n",
       "1  Africa / Mayotte / Mamoudzou  Human                             NaN   \n",
       "2  Africa / Mayotte / Mamoudzou  Human                             NaN   \n",
       "3  Africa / Mayotte / Mamoudzou  Human                             NaN   \n",
       "4  Africa / Mayotte / Mamoudzou  Human                             NaN   \n",
       "\n",
       "   Gender Patient age Patient status   Passage        Specimen  \\\n",
       "0  Female          61      Inpatient  Original             NaN   \n",
       "1  Female           8      Inpatient  Original  Nasopharyngeal   \n",
       "2  Female          49      Inpatient  Original  Nasopharyngeal   \n",
       "3    Male          78      Inpatient  Original  Nasopharyngeal   \n",
       "4    Male          34      Inpatient  Original  Nasopharyngeal   \n",
       "\n",
       "  Additional host information  Lineage Clade  \n",
       "0                         NaN  B.1.160    GH  \n",
       "1                         NaN  B.1.351    GH  \n",
       "2                         NaN  B.1.351    GH  \n",
       "3                         NaN  B.1.351    GH  \n",
       "4                         NaN  B.1.351    GH  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "meta.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['Inpatient', 'Hospitalized', 'Live', 'Isolation',\n",
       "       'Not hospitalized', 'Released', 'Recovered', 'Post mortem',\n",
       "       'Outpatient', 'inpatient', 'uknown', 'Stable in quarantine',\n",
       "       'Deceased', 'outpatient', 'Hospitalised', 'Patient status',\n",
       "       'Discharged', 'Hospitalized (Stable)', 'Symptomatic',\n",
       "       'Asymptomatic', 'Suspected Corona',\n",
       "       'Suspected coronavirus infection', 'ICU', 'Overseas inflow',\n",
       "       'Released, Live', 'Hospitalized (Released)',\n",
       "       'Hospitalized, Asymptomatic, Alive',\n",
       "       'Hospitalized, Discharged, Alive', 'Hospitalized / ICU', 'Mild',\n",
       "       'Fever', 'Non-hospitalized, symptomps: sore throat',\n",
       "       'Pneumonia (chest X-ray)', 'Intensive Care Unit',\n",
       "       'Hospitalized (Deceased)', 'Discharged after recovery',\n",
       "       'No clinical signs', 'Quarantine', 'Facility quarantine',\n",
       "       'Domestic infection',\n",
       "       'Initially hospitalized, but now improved and discharged',\n",
       "       'Hospitalized (Mild)', 'Asymptomatic/Released', 'DAMA', 'Alive',\n",
       "       'Pneumonia (chest X-ray), not critical', 'ICU; Serious', 'Stable',\n",
       "       'Serious', nan, 'Non-hospitalized, symptomps: cough',\n",
       "       'Quarantined', 'Critical', 'Live, mild fever and weakness',\n",
       "       'Non-hospitalized, symptoms: runny nose', 'Mild Sympthoms',\n",
       "       'Pneumonia', 'Undetlying DM and Ischemic Heart Disease',\n",
       "       'hospitalized', 'released', 'Rebased',\n",
       "       'Mild clinical signs without hospitalization',\n",
       "       'Symptoms indicative of upper respiratory infection',\n",
       "       'ICD-10 Disease: B34.2 Coronavirus infection, unspecified',\n",
       "       'ICD-10 Disease: Z11.5 Encounter for screening for other viral diseases',\n",
       "       'ICD-10 Disease: J06.9 Acute upper respiratory infection, unspecified',\n",
       "       'ICD-10 Disease: U07.2 COVID-19, virus not identified',\n",
       "       'ICD-10 Disease: J00-J06 Acute upper respiratory infections',\n",
       "       'ICD-10 Disease: Z20.8 Contact with and exposure to other communicable diseases',\n",
       "       'ICD-10 Disease: Z00.0 Encounter for general adult medical examination',\n",
       "       'Paucisymptomatic', 'Symptomatic, Outpatient',\n",
       "       'Contact surveillance', 'Nursing home', 'unkown', 'EHPAD',\n",
       "       'Screening', 'EHPAD_IRA', 'Home', 'Mild, at home.',\n",
       "       'Live, mild symptoms, at home', 'Culled for pelting',\n",
       "       'ICD-10 Disease: U07.1 COVID-19, virus identified',\n",
       "       'ICD-10 Disease: J06.8 Other acute upper respiratory infections of multiple sites',\n",
       "       'Mild clinical signs without hospitalization. Distorted ability to smell',\n",
       "       'Mild clinical signs without hospitalization. Distorted ability to smell and taste',\n",
       "       'Mild symptomatology', 'Sympotomatic', 'unknonw', 'Ambulatory',\n",
       "       'Hospitalized, no clinical signs', 'Mild clinical signs',\n",
       "       'Mild clinical signs without hospitalization. Distorted ability to smell and taste.',\n",
       "       'Mild clinical signs without hospitalization. Distorted ability to smell.',\n",
       "       'Hospitalized (Intensive care unit and Released)', 'Bronchitis',\n",
       "       'Hospitalized (Live)', 'Symptomatic/Reinfection', 'Mild infection',\n",
       "       'Severe infection', 'Severe', 'Nursing Home',\n",
       "       'Live, physical examination', 'Live, acute respiratory infection',\n",
       "       'ICD-10 Disease: Z03.8 Observation for other suspected diseases and conditions',\n",
       "       'Contact with and exposure to other communicable diseases',\n",
       "       'Acute upper respiratory infection, unspecified',\n",
       "       'Acute bronchitis', 'Pneumonia, unspecified organism',\n",
       "       'Hospitalized (Moderate)', 'Hospitalized (Severe)',\n",
       "       'ICD-10 Disease: Z03.8 Encounter for observation for other suspected diseases and conditions ruled out',\n",
       "       'ICD-10 Disease: Z00.0 Encounter for general adult medical examination without abnormal findings',\n",
       "       'ICD-10 Disease: Z04 Encounter for examination and observation for other reasons',\n",
       "       'ICD-10 Disease: J18.1 Lobar pneumonia, unspecified organism',\n",
       "       'ICD-10 Disease: J18.9 Pneumonia, unspecified organism',\n",
       "       'ICD-10 Disease: U07.2 Virus not identified',\n",
       "       'ICD-10 Disease: Z04.8 Encounter for examination and observation for request for expert evidence',\n",
       "       'Hospitalized, noninvasive ventilation',\n",
       "       'Mild clinical signs without hospitalization. Diarrhoea',\n",
       "       'Nursing house',\n",
       "       'Asymptomatic, identified as positive during preoperation investigation',\n",
       "       'Hospitalized (Intensive care unit)', 'Hospitalized (Critical)',\n",
       "       'ICD-10 Disease: J02.9 Acute pharyngitis, unspecified', 'Cough',\n",
       "       'ICD-10 Disease: J00- Acute nasopharyngitis [common cold]',\n",
       "       'ICD-10 Disease: Z10.8 Routine general health check-up of other defined subpopulations',\n",
       "       'House assistance, Symptomatic', 'unknow', 'Hospitlalized', 'UCI',\n",
       "       'Asymtomatic', 'Paucisymtomatic',\n",
       "       'Mild symptoms (fever, cardiovascular disorders)',\n",
       "       'Hospitalized, oxygenotherapy, diarrhea', 'Coronavirus infection',\n",
       "       'Encounter for general adult medical examination',\n",
       "       'Encounter for observation for other suspected diseases and conditions ruled out',\n",
       "       'Other acute upper respiratory infections of multiple sites',\n",
       "       'Aymptomatic', 'Hospidalized', 'Hospitalizes', 'Out-patient',\n",
       "       'Live, asymptomatic', 'Epidemiology Study',\n",
       "       'Mild clinical signs without hospitalization. Distorted ability to taste',\n",
       "       'deceased', 'unkonwn', 'live', 'unknkown', 'Recovering',\n",
       "       'Hospitalized (possible MIS-C)', 'symptomatic',\n",
       "       'Moderate / Outpatient', 'Mild/Contact exposure/Asymptomatic',\n",
       "       'Severe / ICU', 'Mild / Contact exposure / Asymptomatic', 'Screen',\n",
       "       '51.0', '68.0', '42.0', '55.0', '45.0', '37.0', '103.0', '28.0',\n",
       "       '58.0', '43.0', '18.0', '67.0', '71.0', '44.0', '59.0', '61.0',\n",
       "       '38.0', '54.0', 'Reinfection', 'In-hospital'], dtype=object)"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "meta['Patient status'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "high_risk = {\n",
    "    'hospitalized': meta[(meta['Patient status'].str.lower().str.contains('hospitalized|hospitalizes|hospitlalized', na=False))\n",
    "                         & (~meta['Patient status'].str.lower().str.contains('non|not', na=False))],\n",
    "    'inpatient': meta[meta['Patient status'].str.lower().str.contains('inpatient', na=False)],\n",
    "    'deceased': meta[meta['Patient status'].str.lower().str.contains('deceased', na=False)],\n",
    "    'severe': meta[meta['Patient status'].str.lower().str.contains('severe', na=False)]\n",
    "}\n",
    "\n",
    "low_risk = {\n",
    "    'outpatient': meta[meta['Patient status'].str.lower().str.contains('outpatient', na=False)],\n",
    "    'asymptomatic': meta[meta['Patient status'].str.lower().str.contains('asymptomatic', na=False)],\n",
    "    'mild': meta[meta['Patient status'].str.lower().str.contains('mild', na=False)],\n",
    "    'home': meta[meta['Patient status'].str.lower().str.contains('home|not hospitalized', na=False)]\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "high_risk_df = pd.concat(high_risk.values(), ignore_index=True).drop_duplicates(ignore_index=True)\n",
    "low_risk_df = pd.concat(low_risk.values(), ignore_index=True).drop_duplicates(ignore_index=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "high_risk_df['severe'] = 1\n",
    "low_risk_df['severe'] = 0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "final_meta_df = pd.concat([high_risk_df, low_risk_df])[['Virus name', 'Accession ID', 'Collection date', 'severe']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "final_meta_df['name'] = final_meta_df['Virus name'] + '|' + final_meta_df['Accession ID'] + '|' + final_meta_df['Collection date']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Virus name</th>\n",
       "      <th>Accession ID</th>\n",
       "      <th>Collection date</th>\n",
       "      <th>severe</th>\n",
       "      <th>name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>hCoV-19/Nigeria/Lagos01/2020</td>\n",
       "      <td>EPI_ISL_413550</td>\n",
       "      <td>2020-02-27</td>\n",
       "      <td>1</td>\n",
       "      <td>hCoV-19/Nigeria/Lagos01/2020|EPI_ISL_413550|20...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>hCoV-19/DRC/2363/2020</td>\n",
       "      <td>EPI_ISL_437194</td>\n",
       "      <td>2020-04-14</td>\n",
       "      <td>1</td>\n",
       "      <td>hCoV-19/DRC/2363/2020|EPI_ISL_437194|2020-04-14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>hCoV-19/Nigeria/KW298-CV48/2020</td>\n",
       "      <td>EPI_ISL_487107</td>\n",
       "      <td>2020-05-08</td>\n",
       "      <td>1</td>\n",
       "      <td>hCoV-19/Nigeria/KW298-CV48/2020|EPI_ISL_487107...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>hCoV-19/Ghana/UHAS-K674/2020</td>\n",
       "      <td>EPI_ISL_628760</td>\n",
       "      <td>2020-07-07</td>\n",
       "      <td>1</td>\n",
       "      <td>hCoV-19/Ghana/UHAS-K674/2020|EPI_ISL_628760|20...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>hCoV-19/Ghana/UHAS-H009/2020</td>\n",
       "      <td>EPI_ISL_628761</td>\n",
       "      <td>2020-06-16</td>\n",
       "      <td>1</td>\n",
       "      <td>hCoV-19/Ghana/UHAS-H009/2020|EPI_ISL_628761|20...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                        Virus name    Accession ID Collection date  severe  \\\n",
       "0     hCoV-19/Nigeria/Lagos01/2020  EPI_ISL_413550      2020-02-27       1   \n",
       "1            hCoV-19/DRC/2363/2020  EPI_ISL_437194      2020-04-14       1   \n",
       "2  hCoV-19/Nigeria/KW298-CV48/2020  EPI_ISL_487107      2020-05-08       1   \n",
       "3     hCoV-19/Ghana/UHAS-K674/2020  EPI_ISL_628760      2020-07-07       1   \n",
       "4     hCoV-19/Ghana/UHAS-H009/2020  EPI_ISL_628761      2020-06-16       1   \n",
       "\n",
       "                                                name  \n",
       "0  hCoV-19/Nigeria/Lagos01/2020|EPI_ISL_413550|20...  \n",
       "1    hCoV-19/DRC/2363/2020|EPI_ISL_437194|2020-04-14  \n",
       "2  hCoV-19/Nigeria/KW298-CV48/2020|EPI_ISL_487107...  \n",
       "3  hCoV-19/Ghana/UHAS-K674/2020|EPI_ISL_628760|20...  \n",
       "4  hCoV-19/Ghana/UHAS-H009/2020|EPI_ISL_628761|20...  "
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "final_meta_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "vcf = pd.read_csv('../data/snpeff_fixed_updated.vcf', sep='\\t', skiprows=8, low_memory=False, memory_map=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(15176, 26858)"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>NC_045512.2Severeacuterespiratorysyndromecoronavirus2isolateWuhan-Hu-1,completegenome</th>\n",
       "      <th>...</th>\n",
       "      <th>hCoV-19/Colombia/AMA-INS-VG-632/2021|EPI_ISL_956288|2021-01-13</th>\n",
       "      <th>hCoV-19/Colombia/AMA-INS-VG-633/2021|EPI_ISL_956289|2021-01-12</th>\n",
       "      <th>hCoV-19/Colombia/AMA-INS-VG-634/2021|EPI_ISL_956290|2021-01-12</th>\n",
       "      <th>hCoV-19/Colombia/AMA-INS-VG-635/2021|EPI_ISL_956291|2021-01-22</th>\n",
       "      <th>hCoV-19/Colombia/AMA-INS-VG-636/2021|EPI_ISL_956292|2021-01-22</th>\n",
       "      <th>hCoV-19/Colombia/AMA-INS-VG-637/2021|EPI_ISL_956293|2021-01-20</th>\n",
       "      <th>hCoV-19/Colombia/AMA-INS-VG-638/2021|EPI_ISL_956294|2021-01-18</th>\n",
       "      <th>hCoV-19/Colombia/AMA-INS-VG-367/2021|EPI_ISL_956297|2021-01-04</th>\n",
       "      <th>hCoV-19/Colombia/VAC-INS-RI-011/2021|EPI_ISL_956298|2021-01-13</th>\n",
       "      <th>hCoV-19/Brazil/RS-8680/2021|EPI_ISL_983865|2021-02-01</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NC_045512.2</td>\n",
       "      <td>1</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>*,C,G,T,M</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ANN=C|upstream_gene_variant|MODIFIER|ORF1ab|GU...</td>\n",
       "      <td>GT</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NC_045512.2</td>\n",
       "      <td>2</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>*,C,A,G,K,Y,W</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ANN=A|upstream_gene_variant|MODIFIER|ORF1ab|GU...</td>\n",
       "      <td>GT</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NC_045512.2</td>\n",
       "      <td>3</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>*,C,A,W,G,K</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ANN=A|upstream_gene_variant|MODIFIER|ORF1ab|GU...</td>\n",
       "      <td>GT</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NC_045512.2</td>\n",
       "      <td>4</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>*,G,T,W,C,M</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ANN=C|upstream_gene_variant|MODIFIER|ORF1ab|GU...</td>\n",
       "      <td>GT</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NC_045512.2</td>\n",
       "      <td>5</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>*,T,G,C,M</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ANN=C|upstream_gene_variant|MODIFIER|ORF1ab|GU...</td>\n",
       "      <td>GT</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 26858 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        #CHROM  POS ID REF            ALT QUAL FILTER  \\\n",
       "0  NC_045512.2    1  .   A      *,C,G,T,M    .      .   \n",
       "1  NC_045512.2    2  .   T  *,C,A,G,K,Y,W    .      .   \n",
       "2  NC_045512.2    3  .   T    *,C,A,W,G,K    .      .   \n",
       "3  NC_045512.2    4  .   A    *,G,T,W,C,M    .      .   \n",
       "4  NC_045512.2    5  .   A      *,T,G,C,M    .      .   \n",
       "\n",
       "                                                INFO FORMAT  \\\n",
       "0  ANN=C|upstream_gene_variant|MODIFIER|ORF1ab|GU...     GT   \n",
       "1  ANN=A|upstream_gene_variant|MODIFIER|ORF1ab|GU...     GT   \n",
       "2  ANN=A|upstream_gene_variant|MODIFIER|ORF1ab|GU...     GT   \n",
       "3  ANN=C|upstream_gene_variant|MODIFIER|ORF1ab|GU...     GT   \n",
       "4  ANN=C|upstream_gene_variant|MODIFIER|ORF1ab|GU...     GT   \n",
       "\n",
       "   NC_045512.2Severeacuterespiratorysyndromecoronavirus2isolateWuhan-Hu-1,completegenome  \\\n",
       "0                                                  0                                       \n",
       "1                                                  0                                       \n",
       "2                                                  0                                       \n",
       "3                                                  0                                       \n",
       "4                                                  0                                       \n",
       "\n",
       "   ...  hCoV-19/Colombia/AMA-INS-VG-632/2021|EPI_ISL_956288|2021-01-13  \\\n",
       "0  ...                                                  1                \n",
       "1  ...                                                  1                \n",
       "2  ...                                                  1                \n",
       "3  ...                                                  1                \n",
       "4  ...                                                  1                \n",
       "\n",
       "   hCoV-19/Colombia/AMA-INS-VG-633/2021|EPI_ISL_956289|2021-01-12  \\\n",
       "0                                                  1                \n",
       "1                                                  1                \n",
       "2                                                  1                \n",
       "3                                                  1                \n",
       "4                                                  1                \n",
       "\n",
       "   hCoV-19/Colombia/AMA-INS-VG-634/2021|EPI_ISL_956290|2021-01-12  \\\n",
       "0                                                  1                \n",
       "1                                                  1                \n",
       "2                                                  1                \n",
       "3                                                  1                \n",
       "4                                                  1                \n",
       "\n",
       "   hCoV-19/Colombia/AMA-INS-VG-635/2021|EPI_ISL_956291|2021-01-22  \\\n",
       "0                                                  1                \n",
       "1                                                  1                \n",
       "2                                                  1                \n",
       "3                                                  1                \n",
       "4                                                  1                \n",
       "\n",
       "   hCoV-19/Colombia/AMA-INS-VG-636/2021|EPI_ISL_956292|2021-01-22  \\\n",
       "0                                                  1                \n",
       "1                                                  1                \n",
       "2                                                  1                \n",
       "3                                                  1                \n",
       "4                                                  1                \n",
       "\n",
       "   hCoV-19/Colombia/AMA-INS-VG-637/2021|EPI_ISL_956293|2021-01-20  \\\n",
       "0                                                  1                \n",
       "1                                                  1                \n",
       "2                                                  1                \n",
       "3                                                  1                \n",
       "4                                                  1                \n",
       "\n",
       "   hCoV-19/Colombia/AMA-INS-VG-638/2021|EPI_ISL_956294|2021-01-18  \\\n",
       "0                                                  1                \n",
       "1                                                  1                \n",
       "2                                                  1                \n",
       "3                                                  1                \n",
       "4                                                  1                \n",
       "\n",
       "   hCoV-19/Colombia/AMA-INS-VG-367/2021|EPI_ISL_956297|2021-01-04  \\\n",
       "0                                                  1                \n",
       "1                                                  1                \n",
       "2                                                  1                \n",
       "3                                                  1                \n",
       "4                                                  1                \n",
       "\n",
       "   hCoV-19/Colombia/VAC-INS-RI-011/2021|EPI_ISL_956298|2021-01-13  \\\n",
       "0                                                  1                \n",
       "1                                                  1                \n",
       "2                                                  1                \n",
       "3                                                  1                \n",
       "4                                                  1                \n",
       "\n",
       "   hCoV-19/Brazil/RS-8680/2021|EPI_ISL_983865|2021-02-01  \n",
       "0                                                  1      \n",
       "1                                                  1      \n",
       "2                                                  1      \n",
       "3                                                  1      \n",
       "4                                                  1      \n",
       "\n",
       "[5 rows x 26858 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "ids = set(final_meta_df['Accession ID'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "8921"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(ids)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "index0 = vcf.columns.str.split('|').str[0]\n",
    "index1 = vcf.columns.str.split('|').str[1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['#CHROM', 'POS', 'ID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'FORMAT',\n",
       "       'NC_045512.2Severeacuterespiratorysyndromecoronavirus2isolateWuhan-Hu-1,completegenome'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "index0[:10]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_cols = list(index0[:10]) + list(index1[10:])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "vcf.columns = new_cols"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>NC_045512.2Severeacuterespiratorysyndromecoronavirus2isolateWuhan-Hu-1,completegenome</th>\n",
       "      <th>...</th>\n",
       "      <th>EPI_ISL_956288</th>\n",
       "      <th>EPI_ISL_956289</th>\n",
       "      <th>EPI_ISL_956290</th>\n",
       "      <th>EPI_ISL_956291</th>\n",
       "      <th>EPI_ISL_956292</th>\n",
       "      <th>EPI_ISL_956293</th>\n",
       "      <th>EPI_ISL_956294</th>\n",
       "      <th>EPI_ISL_956297</th>\n",
       "      <th>EPI_ISL_956298</th>\n",
       "      <th>EPI_ISL_983865</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NC_045512.2</td>\n",
       "      <td>1</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>*,C,G,T,M</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ANN=C|upstream_gene_variant|MODIFIER|ORF1ab|GU...</td>\n",
       "      <td>GT</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NC_045512.2</td>\n",
       "      <td>2</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>*,C,A,G,K,Y,W</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ANN=A|upstream_gene_variant|MODIFIER|ORF1ab|GU...</td>\n",
       "      <td>GT</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NC_045512.2</td>\n",
       "      <td>3</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>*,C,A,W,G,K</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ANN=A|upstream_gene_variant|MODIFIER|ORF1ab|GU...</td>\n",
       "      <td>GT</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NC_045512.2</td>\n",
       "      <td>4</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>*,G,T,W,C,M</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ANN=C|upstream_gene_variant|MODIFIER|ORF1ab|GU...</td>\n",
       "      <td>GT</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NC_045512.2</td>\n",
       "      <td>5</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>*,T,G,C,M</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ANN=C|upstream_gene_variant|MODIFIER|ORF1ab|GU...</td>\n",
       "      <td>GT</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15171</th>\n",
       "      <td>NC_045512.2</td>\n",
       "      <td>29899</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>*,C,T,G,H,M</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ANN=C|downstream_gene_variant|MODIFIER|S|GU280...</td>\n",
       "      <td>GT</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15172</th>\n",
       "      <td>NC_045512.2</td>\n",
       "      <td>29900</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>*,T,G,C,M,W</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ANN=C|downstream_gene_variant|MODIFIER|S|GU280...</td>\n",
       "      <td>GT</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15173</th>\n",
       "      <td>NC_045512.2</td>\n",
       "      <td>29901</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>*,T,G,C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ANN=C|downstream_gene_variant|MODIFIER|S|GU280...</td>\n",
       "      <td>GT</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15174</th>\n",
       "      <td>NC_045512.2</td>\n",
       "      <td>29902</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>*,T,C,G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ANN=C|downstream_gene_variant|MODIFIER|S|GU280...</td>\n",
       "      <td>GT</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15175</th>\n",
       "      <td>NC_045512.2</td>\n",
       "      <td>29903</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>*,T,C,G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ANN=C|downstream_gene_variant|MODIFIER|S|GU280...</td>\n",
       "      <td>GT</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>15176 rows × 26858 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            #CHROM    POS ID REF            ALT QUAL FILTER  \\\n",
       "0      NC_045512.2      1  .   A      *,C,G,T,M    .      .   \n",
       "1      NC_045512.2      2  .   T  *,C,A,G,K,Y,W    .      .   \n",
       "2      NC_045512.2      3  .   T    *,C,A,W,G,K    .      .   \n",
       "3      NC_045512.2      4  .   A    *,G,T,W,C,M    .      .   \n",
       "4      NC_045512.2      5  .   A      *,T,G,C,M    .      .   \n",
       "...            ...    ... ..  ..            ...  ...    ...   \n",
       "15171  NC_045512.2  29899  .   A    *,C,T,G,H,M    .      .   \n",
       "15172  NC_045512.2  29900  .   A    *,T,G,C,M,W    .      .   \n",
       "15173  NC_045512.2  29901  .   A        *,T,G,C    .      .   \n",
       "15174  NC_045512.2  29902  .   A        *,T,C,G    .      .   \n",
       "15175  NC_045512.2  29903  .   A        *,T,C,G    .      .   \n",
       "\n",
       "                                                    INFO FORMAT  \\\n",
       "0      ANN=C|upstream_gene_variant|MODIFIER|ORF1ab|GU...     GT   \n",
       "1      ANN=A|upstream_gene_variant|MODIFIER|ORF1ab|GU...     GT   \n",
       "2      ANN=A|upstream_gene_variant|MODIFIER|ORF1ab|GU...     GT   \n",
       "3      ANN=C|upstream_gene_variant|MODIFIER|ORF1ab|GU...     GT   \n",
       "4      ANN=C|upstream_gene_variant|MODIFIER|ORF1ab|GU...     GT   \n",
       "...                                                  ...    ...   \n",
       "15171  ANN=C|downstream_gene_variant|MODIFIER|S|GU280...     GT   \n",
       "15172  ANN=C|downstream_gene_variant|MODIFIER|S|GU280...     GT   \n",
       "15173  ANN=C|downstream_gene_variant|MODIFIER|S|GU280...     GT   \n",
       "15174  ANN=C|downstream_gene_variant|MODIFIER|S|GU280...     GT   \n",
       "15175  ANN=C|downstream_gene_variant|MODIFIER|S|GU280...     GT   \n",
       "\n",
       "       NC_045512.2Severeacuterespiratorysyndromecoronavirus2isolateWuhan-Hu-1,completegenome  \\\n",
       "0                                                      0                                       \n",
       "1                                                      0                                       \n",
       "2                                                      0                                       \n",
       "3                                                      0                                       \n",
       "4                                                      0                                       \n",
       "...                                                  ...                                       \n",
       "15171                                                  0                                       \n",
       "15172                                                  0                                       \n",
       "15173                                                  0                                       \n",
       "15174                                                  0                                       \n",
       "15175                                                  0                                       \n",
       "\n",
       "       ...  EPI_ISL_956288  EPI_ISL_956289  EPI_ISL_956290  EPI_ISL_956291  \\\n",
       "0      ...               1               1               1               1   \n",
       "1      ...               1               1               1               1   \n",
       "2      ...               1               1               1               1   \n",
       "3      ...               1               1               1               1   \n",
       "4      ...               1               1               1               1   \n",
       "...    ...             ...             ...             ...             ...   \n",
       "15171  ...               1               1               1               1   \n",
       "15172  ...               1               1               1               1   \n",
       "15173  ...               1               1               1               1   \n",
       "15174  ...               1               1               1               1   \n",
       "15175  ...               1               1               1               1   \n",
       "\n",
       "       EPI_ISL_956292  EPI_ISL_956293  EPI_ISL_956294  EPI_ISL_956297  \\\n",
       "0                   1               1               1               1   \n",
       "1                   1               1               1               1   \n",
       "2                   1               1               1               1   \n",
       "3                   1               1               1               1   \n",
       "4                   1               1               1               1   \n",
       "...               ...             ...             ...             ...   \n",
       "15171               1               1               1               1   \n",
       "15172               1               1               1               1   \n",
       "15173               1               1               1               1   \n",
       "15174               1               1               1               1   \n",
       "15175               1               1               1               1   \n",
       "\n",
       "       EPI_ISL_956298  EPI_ISL_983865  \n",
       "0                   1               1  \n",
       "1                   1               1  \n",
       "2                   1               1  \n",
       "3                   1               1  \n",
       "4                   1               1  \n",
       "...               ...             ...  \n",
       "15171               1               1  \n",
       "15172               1               1  \n",
       "15173               1               1  \n",
       "15174               1               1  \n",
       "15175               1               1  \n",
       "\n",
       "[15176 rows x 26858 columns]"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "vcf_subset = vcf.loc[:, vcf.columns.isin(ids)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>EPI_ISL_455412</th>\n",
       "      <th>EPI_ISL_455413</th>\n",
       "      <th>EPI_ISL_455415</th>\n",
       "      <th>EPI_ISL_455418</th>\n",
       "      <th>EPI_ISL_455419</th>\n",
       "      <th>EPI_ISL_455422</th>\n",
       "      <th>EPI_ISL_455423</th>\n",
       "      <th>EPI_ISL_455424</th>\n",
       "      <th>EPI_ISL_455426</th>\n",
       "      <th>EPI_ISL_455427</th>\n",
       "      <th>...</th>\n",
       "      <th>EPI_ISL_942930</th>\n",
       "      <th>EPI_ISL_942931</th>\n",
       "      <th>EPI_ISL_954090</th>\n",
       "      <th>EPI_ISL_954094</th>\n",
       "      <th>EPI_ISL_954095</th>\n",
       "      <th>EPI_ISL_954096</th>\n",
       "      <th>EPI_ISL_954097</th>\n",
       "      <th>EPI_ISL_954098</th>\n",
       "      <th>EPI_ISL_954099</th>\n",
       "      <th>EPI_ISL_983865</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15171</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15172</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15173</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15174</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15175</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>15176 rows × 8914 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       EPI_ISL_455412  EPI_ISL_455413  EPI_ISL_455415  EPI_ISL_455418  \\\n",
       "0                   1               1               1               1   \n",
       "1                   1               1               1               1   \n",
       "2                   1               1               1               1   \n",
       "3                   1               1               1               1   \n",
       "4                   1               1               1               1   \n",
       "...               ...             ...             ...             ...   \n",
       "15171               1               1               1               1   \n",
       "15172               1               1               1               1   \n",
       "15173               1               1               1               1   \n",
       "15174               1               1               1               1   \n",
       "15175               1               1               1               1   \n",
       "\n",
       "       EPI_ISL_455419  EPI_ISL_455422  EPI_ISL_455423  EPI_ISL_455424  \\\n",
       "0                   1               1               1               1   \n",
       "1                   1               1               1               1   \n",
       "2                   1               1               1               1   \n",
       "3                   1               1               1               1   \n",
       "4                   1               1               1               1   \n",
       "...               ...             ...             ...             ...   \n",
       "15171               1               1               1               1   \n",
       "15172               1               1               1               1   \n",
       "15173               1               1               1               1   \n",
       "15174               1               1               1               1   \n",
       "15175               1               1               1               1   \n",
       "\n",
       "       EPI_ISL_455426  EPI_ISL_455427  ...  EPI_ISL_942930  EPI_ISL_942931  \\\n",
       "0                   1               1  ...               1               1   \n",
       "1                   1               1  ...               1               1   \n",
       "2                   1               1  ...               0               1   \n",
       "3                   1               1  ...               0               1   \n",
       "4                   1               1  ...               0               1   \n",
       "...               ...             ...  ...             ...             ...   \n",
       "15171               1               1  ...               1               1   \n",
       "15172               1               1  ...               1               1   \n",
       "15173               1               1  ...               1               1   \n",
       "15174               1               1  ...               1               1   \n",
       "15175               1               1  ...               1               1   \n",
       "\n",
       "       EPI_ISL_954090  EPI_ISL_954094  EPI_ISL_954095  EPI_ISL_954096  \\\n",
       "0                   1               1               1               1   \n",
       "1                   1               1               1               1   \n",
       "2                   1               1               1               1   \n",
       "3                   1               1               1               1   \n",
       "4                   1               1               1               1   \n",
       "...               ...             ...             ...             ...   \n",
       "15171               1               1               1               1   \n",
       "15172               1               1               1               1   \n",
       "15173               1               1               1               1   \n",
       "15174               1               1               1               1   \n",
       "15175               1               1               1               1   \n",
       "\n",
       "       EPI_ISL_954097  EPI_ISL_954098  EPI_ISL_954099  EPI_ISL_983865  \n",
       "0                   1               1               1               1  \n",
       "1                   1               1               1               1  \n",
       "2                   1               1               1               1  \n",
       "3                   1               1               1               1  \n",
       "4                   1               1               1               1  \n",
       "...               ...             ...             ...             ...  \n",
       "15171               1               1               1               1  \n",
       "15172               1               1               1               1  \n",
       "15173               1               1               1               1  \n",
       "15174               1               1               1               1  \n",
       "15175               1               1               1               1  \n",
       "\n",
       "[15176 rows x 8914 columns]"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf_subset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "severe_acc = set(final_meta_df.loc[final_meta_df['severe'] == 1]['Accession ID'])\n",
    "mild_acc = set(final_meta_df.loc[final_meta_df['severe'] == 0]['Accession ID'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "var_id = []\n",
    "severe_tot = []\n",
    "mild_tot = []\n",
    "frequency = []\n",
    "var_type = []\n",
    "var_impact = []\n",
    "var_gene = []"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[============================================================] 100.1% \r"
     ]
    }
   ],
   "source": [
    "for i,row in vcf_subset.iterrows():\n",
    "    progress(i, len(vcf_subset))\n",
    "    \n",
    "    for num,var in enumerate(vcf.iloc[i]['ALT']):\n",
    "        severe_num = 0\n",
    "        mild_num = 0\n",
    "        \n",
    "        all_anns = [v.split('|')[0] for v in vcf.iloc[i]['INFO'].split(',')]\n",
    "        all_anns[0] = all_anns[0].replace('ANN=', '')\n",
    "          \n",
    "        if var in ['A', 'T', 'C', 'G']:\n",
    "            index = all_anns.index(var)\n",
    "            var_id.append('{0}{1}>{2}'.format(vcf.iloc[i]['POS'], vcf.iloc[i]['REF'], var))\n",
    "            var_type.append(vcf.iloc[i]['INFO'].split(',')[index].split('|')[1])\n",
    "            var_impact.append(vcf.iloc[i]['INFO'].split(',')[index].split('|')[2])\n",
    "            var_gene.append(vcf.iloc[i]['INFO'].split(',')[index].split('|')[3])\n",
    "            \n",
    "            frequency.append(len(vcf_subset.columns[(vcf_subset == num + 1).iloc[i]]) / len(vcf_subset.iloc[i]))\n",
    "            \n",
    "            for col in vcf_subset.columns[(vcf_subset == num + 1).iloc[i]]:\n",
    "                if col in severe_acc:\n",
    "                    severe_num += 1\n",
    "                else:\n",
    "                    mild_num += 1\n",
    "                    \n",
    "            severe_tot.append(severe_num)\n",
    "            mild_tot.append(mild_num)\n",
    "            \n",
    "        else:\n",
    "            continue"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "final_df = pd.DataFrame({'variant': var_id, 'num_severe': severe_tot, 'num_mild': mild_tot, 'var_type': var_type, \n",
    "                         'frequency': frequency, 'impact': var_impact, 'gene': var_gene})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [],
   "source": [
    "final_df['num_freq'] = final_df['num_severe'] + final_df['num_mild']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "final_df = final_df.loc[final_df['num_freq'] != 0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>variant</th>\n",
       "      <th>num_severe</th>\n",
       "      <th>num_mild</th>\n",
       "      <th>var_type</th>\n",
       "      <th>frequency</th>\n",
       "      <th>impact</th>\n",
       "      <th>gene</th>\n",
       "      <th>num_freq</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1A&gt;C</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>0.000224</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>ORF1ab</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2T&gt;C</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>0.000449</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>ORF1ab</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>3T&gt;C</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>0.000337</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>ORF1ab</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>3T&gt;A</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>0.000224</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>ORF1ab</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>4A&gt;G</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>0.000897</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>ORF1ab</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17896</th>\n",
       "      <td>29898A&gt;T</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>0.000449</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>S</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17899</th>\n",
       "      <td>29899A&gt;C</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>0.000112</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>S</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17902</th>\n",
       "      <td>29900A&gt;T</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>0.000337</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>S</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17905</th>\n",
       "      <td>29901A&gt;T</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>0.000337</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>S</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17908</th>\n",
       "      <td>29902A&gt;T</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>0.000224</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>S</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3389 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        variant  num_severe  num_mild                 var_type  frequency  \\\n",
       "0          1A>C           2         0    upstream_gene_variant   0.000224   \n",
       "3          2T>C           3         1    upstream_gene_variant   0.000449   \n",
       "6          3T>C           3         0    upstream_gene_variant   0.000337   \n",
       "7          3T>A           1         1    upstream_gene_variant   0.000224   \n",
       "9          4A>G           8         0    upstream_gene_variant   0.000897   \n",
       "...         ...         ...       ...                      ...        ...   \n",
       "17896  29898A>T           3         1  downstream_gene_variant   0.000449   \n",
       "17899  29899A>C           1         0  downstream_gene_variant   0.000112   \n",
       "17902  29900A>T           3         0  downstream_gene_variant   0.000337   \n",
       "17905  29901A>T           3         0  downstream_gene_variant   0.000337   \n",
       "17908  29902A>T           2         0  downstream_gene_variant   0.000224   \n",
       "\n",
       "         impact    gene  num_freq  \n",
       "0      MODIFIER  ORF1ab         2  \n",
       "3      MODIFIER  ORF1ab         4  \n",
       "6      MODIFIER  ORF1ab         3  \n",
       "7      MODIFIER  ORF1ab         2  \n",
       "9      MODIFIER  ORF1ab         8  \n",
       "...         ...     ...       ...  \n",
       "17896  MODIFIER       S         4  \n",
       "17899  MODIFIER       S         1  \n",
       "17902  MODIFIER       S         3  \n",
       "17905  MODIFIER       S         3  \n",
       "17908  MODIFIER       S         2  \n",
       "\n",
       "[3389 rows x 8 columns]"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "final_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "final_df.to_csv('../data/variants.csv', index=False, header=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [],
   "source": [
    "a = final_df.sample(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>variant</th>\n",
       "      <th>num_severe</th>\n",
       "      <th>num_mild</th>\n",
       "      <th>var_type</th>\n",
       "      <th>frequency</th>\n",
       "      <th>impact</th>\n",
       "      <th>gene</th>\n",
       "      <th>num_freq</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>9185</th>\n",
       "      <td>16733C&gt;A</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>0.000224</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>ORF1ab</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1808</th>\n",
       "      <td>2275A&gt;G</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>0.000112</td>\n",
       "      <td>LOW</td>\n",
       "      <td>ORF1ab</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5327</th>\n",
       "      <td>8334C&gt;T</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>0.000897</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>ORF1ab</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9376</th>\n",
       "      <td>17122G&gt;T</td>\n",
       "      <td>13</td>\n",
       "      <td>0</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>0.001458</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>ORF1ab</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7485</th>\n",
       "      <td>12876C&gt;A</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>0.000224</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>ORF1ab</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15622</th>\n",
       "      <td>27678A&gt;G</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>0.000112</td>\n",
       "      <td>LOW</td>\n",
       "      <td>ORF7a</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12928</th>\n",
       "      <td>23611G&gt;T</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>0.000112</td>\n",
       "      <td>LOW</td>\n",
       "      <td>S</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14219</th>\n",
       "      <td>25710C&gt;T</td>\n",
       "      <td>384</td>\n",
       "      <td>312</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>0.078079</td>\n",
       "      <td>LOW</td>\n",
       "      <td>ORF3a</td>\n",
       "      <td>696</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>912</th>\n",
       "      <td>922G&gt;T</td>\n",
       "      <td>20</td>\n",
       "      <td>0</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>0.002244</td>\n",
       "      <td>LOW</td>\n",
       "      <td>ORF1ab</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7989</th>\n",
       "      <td>14033A&gt;G</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>0.000112</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>ORF1ab</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        variant  num_severe  num_mild            var_type  frequency  \\\n",
       "9185   16733C>A           2         0         stop_gained   0.000224   \n",
       "1808    2275A>G           1         0  synonymous_variant   0.000112   \n",
       "5327    8334C>T           8         0    missense_variant   0.000897   \n",
       "9376   17122G>T          13         0    missense_variant   0.001458   \n",
       "7485   12876C>A           2         0    missense_variant   0.000224   \n",
       "15622  27678A>G           0         1  synonymous_variant   0.000112   \n",
       "12928  23611G>T           1         0  synonymous_variant   0.000112   \n",
       "14219  25710C>T         384       312  synonymous_variant   0.078079   \n",
       "912      922G>T          20         0  synonymous_variant   0.002244   \n",
       "7989   14033A>G           1         0    missense_variant   0.000112   \n",
       "\n",
       "         impact    gene  num_freq  \n",
       "9185       HIGH  ORF1ab         2  \n",
       "1808        LOW  ORF1ab         1  \n",
       "5327   MODERATE  ORF1ab         8  \n",
       "9376   MODERATE  ORF1ab        13  \n",
       "7485   MODERATE  ORF1ab         2  \n",
       "15622       LOW   ORF7a         1  \n",
       "12928       LOW       S         1  \n",
       "14219       LOW   ORF3a       696  \n",
       "912         LOW  ORF1ab        20  \n",
       "7989   MODERATE  ORF1ab         1  "
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "a"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [],
   "source": [
    "final_df.to_csv('../data/variants_10.csv', index=False, header=True)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
